core/src/main/scala/shapeless/generic.scala

```diff
@@ -698,6 +723,8 @@ trait CaseClassMacros extends ReprTypes with CaseClassMacrosVersionSpecifics {
     mkAttributedRef(pre, getter)
   }
 
+  /** Check if `sym` is annotated by [[nonGeneric]]
+   *  including more processes to avoid known issues about annotation and macro. */
```

Comment:
Sorry I didn't notice this before. I think the callers of this method don't care about the workarounds for bugs.
```suggestion
  /**
   * Check if `sym` or any of its overrides are annotated by [[nonGeneric]]
   */
```